cache/cache.go

```diff
@@ -181,86 +168,41 @@ func (c *cache[T]) set(key string, object T) {
 	c.items = append(c.items, &item)
 }
 
-// Get an item from the cache. Returns the item or nil, and a bool indicating
-// whether the key was found.
-func (c *Cache[T]) Get(object T) (item T, exists bool, err error) {
-	var res T
-	lvs := []string{}
-	if c.labelsFunc != nil {
-		lvs, err = c.labelsFunc(object, len(c.metrics.getExtraLabels()))
-		if err != nil {
-			recordRequest(c.metrics, StatusFailure)
-			return res, false, &CacheError{Reason: ErrInvalidLabels, Err: err}
-		}
-	}
-	key, err := c.keyFunc(object)
-	if err != nil {
-		recordRequest(c.metrics, StatusFailure)
-		return res, false, &CacheError{Reason: ErrInvalidKey, Err: err}
-	}
-	item, found, err := c.get(key)
-	if err != nil {
-		return res, false, err
-	}
-	if !found {
-		recordEvent(c.metrics, CacheEventTypeMiss, lvs...)
-		return res, false, nil
-	}
-	recordEvent(c.metrics, CacheEventTypeHit, lvs...)
-	return item, true, nil
-}
-
-// GetByKey returns the object for the given key.
-func (c *Cache[T]) GetByKey(key string) (T, bool, error) {
-	var res T
-	index, found, err := c.get(key)
-	if err != nil {
-		return res, false, err
-	}
-	if !found {
-		recordEvent(c.metrics, CacheEventTypeMiss)
-		return res, false, nil
-	}
-
-	recordEvent(c.metrics, CacheEventTypeHit)
-	return index, true, nil
-}
-
-func (c *cache[T]) get(key string) (T, bool, error) {
-	var res T
+// Get returns a pointer to an item in the cache for the given key. If no item
+// is found, it's a nil pointer.
+// The caller can record cache hit or miss based on the result with
+// Cache.RecordCacheEvent().
+func (c *Cache[T]) Get(key string) (*T, error) {
 	c.mu.RLock()
 	if c.closed {
 		c.mu.RUnlock()
 		recordRequest(c.metrics, StatusFailure)
-		return res, false, ErrCacheClosed
+		return nil, ErrCacheClosed
 	}
 	item, found := c.index[key]
 	if !found {
 		c.mu.RUnlock()
 		recordRequest(c.metrics, StatusSuccess)
-		return res, false, nil
+		return nil, nil
 	}
 	if !item.expiresAt.IsZero() {
 		if item.expiresAt.Compare(time.Now()) < 0 {
 			c.mu.RUnlock()
 			recordRequest(c.metrics, StatusSuccess)
-			return res, false, nil
+			return nil, nil
```

Comment:
Return an explicit error here since the caller cannot figure out if the item has expired and that value is not being returned.